python/paddle/fluid/layers/io.py

```diff
@@ -470,6 +471,153 @@ def random_data_generator(low, high, shapes, lod_levels, for_parallel=True):
     return monkey_patch_reader_methods(main_prog_var)
 
 
+def _py_reader(capacity,
+               shapes,
+               dtypes,
+               lod_levels=None,
+               name=None,
+               use_double_buffer=True,
+               feed_list=None):
+    if feed_list is not None:
+        assert isinstance(feed_list, list)
+        lod_levels = []
+        dtypes = []
+        shape_concat = []
+        ranks = []
+        shapes = []
+
+        for data in feed_list:
+            dtypes.append(data.dtype)
+            shape_concat.extend(data.shape)
+            ranks.append(len(data.shape))
+            shapes.append(data.shape)
+            lod_levels.append(data.lod_level)
```

Comment:
will names come from data?

Reply:
yes, the variable for data is created by data list